cf-execd/exec-config.c

```diff
@@ -51,6 +51,59 @@ static char *GetIpAddresses(const EvalContext *ctx)
     return StringWriterClose(ipbuf);
 }
 
+static void RegexFree(void *ptr)
+{
+    pcre_free(ptr);
+}
+
+static void MailFilterFill(const void *input, bool rlist,
```

Comment:
This is rather matter of taste, but I would add two functions that you can call from inside ```FillList``` and ```FillSeq```. Having this you will avoid checking ```rlist``` condition over and over again.